lib/Service/MailManager.php

```diff
@@ -292,17 +291,17 @@ public function flagMessage(Account $account, string $mailbox, int $uid, string
 		}
 
 		// Only send system flags to the IMAP server as other flags might not be supported
-		if (in_array(strtolower("\\$flag"), self::ALLOWED_FLAGS, true)) {
-			try {
-				$normalized = '\\' . strtolower($flag);
+		$imapFlags = self::ALLOWED_FLAGS[$flag] ?? [];
+		try {
+			foreach ($imapFlags as $imapFlag) {
 				if ($value) {
-					$this->messageMapper->addFlag($client, $mb, $uid, $normalized);
+					$this->messageMapper->addFlag($client, $mb, $uid, $imapFlag);
```

Comment:
`\` is removed from flag?
It's now `$junk` instead of `\$junk`?

Reply:
No, only registered flags start with `\`. `$Junk` is a keyword.

Reply:
To add to the confusion: we do mix the terms flags/keywords here. Ideally we could name `junk` keyword, but that would probably just complicate the code and lead to two otherwise identical code paths.

Reply:
So the old `$normalized = '\\' . strtolower($flag);` didn't make sense?

Reply:
Only for the flags, yeah. Thus is was fine until now where we add the first keyword. If we set `$Junk` as `\$Junk` the server might just raise an error. Or if it does allow the flag then it has a wrong meaning.